flink-mesos/src/main/java/org/apache/flink/mesos/runtime/clusterframework/LaunchableMesosWorker.java

```diff
@@ -262,9 +265,67 @@ public String toString() {
 			taskInfo.setContainer(containerInfo);
 		}
 
+		containerInfo.addAllVolumes(volumes(params.containerVolumes()));
+
 		return taskInfo.build();
 	}
 
+	/**
+	 * Used to build volume specs for mesos. This allows for mounting additional volumes into a container
+	 *
+	 * @param containerVolumes a comma delimited optional string of [host_path:]container_path[:RO|RW] that
+	 *                         defines mount points for a container volume. If None or empty string, returns
+	 *                         an empty iterator
+	 */
+	public List<Protos.Volume> volumes(Option<String> containerVolumes) {
+		if (containerVolumes.isEmpty()) {
+			return new ArrayList<Protos.Volume>();
+		}
+		String[] specs = containerVolumes.get().split(",");
+		List<Protos.Volume> vols = new ArrayList<Protos.Volume>();
+		for (int i = 0; i < specs.length; i++) {
+			String s = specs[i];
+			if (s.isEmpty()) {
+				continue;
+			}
+			Protos.Volume.Builder vol = Protos.Volume.newBuilder();
+			vol.setMode(Protos.Volume.Mode.RW);
+
+			List<String> parts = Arrays.asList(s.split(":"));
```

Comment:
eh nvm; a messed up mode value will always cause issues. It's just that we can actually detect it if parts.size == 3.